drivers/i2s/i2s_nrfx.c

```diff
@@ -656,15 +676,17 @@ static int start_transfer(struct i2s_nrfx_drv_data *drv_data)
 		/* Failed to allocate next RX buffer */
 		ret = -ENOMEM;
 	} else {
-		uint32_t block_size = (drv_data->active_dir == I2S_DIR_TX)
-				      ? drv_data->tx.cfg.block_size
-				      : drv_data->rx.cfg.block_size;
 		nrfx_err_t err;
 
+		if (drv_data->active_dir == I2S_DIR_RX) {
+			initial_buffers.buffer_size =
+				drv_data->rx.cfg.block_size / sizeof(uint32_t);
+		}
```

Comment:
This way `initial_buffers.buffer_size` will stay initialized to 0 for `I2S_DIR_TX` and `I2S_DIR_BOTH`.
I think you should rather keep `block_size` initialized as it was and here do:
```
initial_buffers.buffer_size = block_size / sizeof(uint32_t);
```

Reply:
No, `initial_buffers.buffer_size` won't stay initialized to 0 for `I2S_DIR_TX` and `I2S_DIR_BOTH`. If you have `I2S_DIR_TX` or `I2S_DIR_BOTH` then `get_next_tx_buffer()` will set `buffers->buffer_size` because it is called just a few lines above in this function (if no buffer is available, then this function will log "No TX buffer available" and return `-ENOMEM`).

The only reason why `initial_buffers.buffer_size` is initialized here is because `get_next_tx_buffer(drv_data, &initial_buffers)` is not called (due to right part of `&&` being short circuited when `active_dir` is `I2S_DIR_RX`).

Reply:
Similar comment could be also helpful in `supply_next_buffers`.